Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ExchangeManagerBean.java

```diff
@@ -327,13 +329,26 @@ public boolean deleteExchange() {
             if (bDelete) {
                 modifiedExchangesCache();
                 selectedExchange = null;
+                filterExchange = null;
+                filterOrganization = null;
+                orgFilter = null;
+                endpoints = null;
+                refreshOrganizations();
             }
         }
         return bDelete;
     }
 
     public int pingEndpoint() {
-        return exchangeService.pingService(selectedEndpoint);
+        String sName = selectedEndpoint.getName();
+        int pingResponse = 0;
```

Comment:
we never set this anywhere in the code. Either remove it and hardcode 0 as a return value or set pingResponse to 0 and break from the loop on line 348.